### PR TITLE
Add org.freedesktop.Sdk.Extension.node22

### PR DESCRIPTION
**Pull request by razzeee**
### Please confirm your submission meets all the criteria

- [x] Please describe the application briefly.
  Port of Node 20 to Node 22

I fail to build this locally, as it runs out of space on both of my main systems.

- [ ] The domain used for the application ID is [controlled by the application developers][appid-domain] and the [application id guidelines][appid] are followed.
- [x] I have read the and followed all the [Submission and licence requirements][reqs].
- [ ] I have [built][build] and tested the submission locally.
- [ ] I am an author/developer/upstream contributor of the project. If not, I contacted upstream developers about this submission. **Link:**

[appid-domain]: https://docs.flathub.org/docs/for-app-authors/requirements/#control-over-domain-or-repository
[appid]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id
[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[build]: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission

### CODE REVIEW

File: org.freedesktop.Sdk.Extension.node22.yaml [hunk 1]
```diff
@@ -0,0 +1,82 @@
+app-id: org.freedesktop.Sdk.Extension.node22
+branch: "23.08"
```
Comment (bbhtt) [inline org.freedesktop.Sdk.Extension.node22.yaml:2]:
```suggestion
branch: "24.08"
```

Comment (bbhtt) [reply org.freedesktop.Sdk.Extension.node22.yaml:2]: 23.08 is oldstable now, new extensions should target 24.08

File: org.freedesktop.Sdk.Extension.node22.yaml [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+runtime: org.freedesktop.Sdk
+runtime-version: "23.08"
+sdk: org.freedesktop.Sdk
+build-extension: true
+separate-locales: false
+appstream-compose: false
```
Comment (bbhtt) [inline org.freedesktop.Sdk.Extension.node22.yaml:8]:
```suggestion
```

File: org.freedesktop.Sdk.Extension.node22.yaml [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+            .tag_name | sub("^[vV]"; "")
+          url-query: >-
+            .assets | map(select(.name|endswith(".tar.gz"))) | first | .browser_download_url
+
+  - name: scripts
+    buildsystem: simple
+    build-commands:
+      - install -Dm755 {enable,install{,-sdk}}.sh -t ${FLATPAK_DEST}/
+      - install -Dm644 ${FLATPAK_ID}.metainfo.xml -t ${FLATPAK_DEST}/share/metainfo
+      - appstream-compose --basename=${FLATPAK_ID} --prefix=${FLATPAK_DEST} --origin=flatpak
```
Comment (bbhtt) [inline org.freedesktop.Sdk.Extension.node22.yaml:65]:
```suggestion
```

**Comment by razzeee** [comment]
Updated accordingly

**Comment by bbhtt** [comment]
bot, build org.freedesktop.Sdk.Extension.node22

**Pull request closed without merging**